uktrade/data-hub-frontend
# Re-write edit client relationship management form in React

## Description of change

This PR re-writes the form to edit an investment project's client relationship manager in react.

This is due to the previous implementation using the vue.js typeahead, and part of a wider task to remove vue.js from the codebase.

Includes:
- Removing old logic from express
- Removing unit test for form/controller
- Adding a functional test spec for the form
- Moving react project team code to the `investments/client/projects/team` folder
- Adding client-side validation that prevents you removing a client relationship manager without adding a new one (previously a user was able to remove it and save the form, however this is a mandatory field in the API for a project so the change was not actually made)

## Test instructions

Go to an investment project
- Go to the project team tab
- Edit the client relationship manager
- You should be able to update the CRM and save the form
- You should not be able to save the form if there is no CRM

## Screenshots
### Before

![Screenshot 2022-02-28 at 14 29 31](https://user-images.githubusercontent.com/22460823/156000289-79410e89-2835-4d2d-958f-33918101062e.png)


### After

![Screenshot 2022-02-28 at 14 29 17](https://user-images.githubusercontent.com/22460823/156000252-194934dc-3ea2-4e33-8a6e-1e46fc82b2e4.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
